object-singleton.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
```

Comment:
Do you want it to be commented?

Reply:
As the code is explained in full in the blog post, do you think its also worth having comments in the code? I guess if people land directly on this code base, then comments might be usefull. Ill add some